plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReloader.java

```diff
@@ -0,0 +1,331 @@
+
+                    autoReloadModelByNodeAndModelId(localNodeId, mlTask.getModelId());
+
+                    // if reload the model successfully,the number of unsuccessful reload should be reset to zero.
+                    retryTimes = 0;
+                } catch (MLException e) {
+                    retryTimes++;
+                    log.error("Can't auto reload model in node id {} ,has tried {} times\nThe reason is:{}", localNodeId, retryTimes, e);
+                }
+
+                // Store the latest value of the retryTimes and node id under the index ".plugins-ml-model-reload"
+                saveLatestRetryTimes(
+                    localNodeId,
+                    retryTimes,
+                    ActionListener.wrap(saveLatestRetryTimesStep::onResponse, saveLatestRetryTimesStep::onFailure)
+                );
+            }, getRetryTimesStep::onFailure);
+        }, queryTaskStep::onFailure);
+
+        saveLatestRetryTimesStep.whenComplete(response -> log.info("successfully complete all steps"), saveLatestRetryTimesStep::onFailure);
+    }
+
+    /**
+     * auto reload 1 model under the node id
+     *
+     * @param localNodeId node id
+     * @param modelId     model id
+     */
+    @VisibleForTesting
+    void autoReloadModelByNodeAndModelId(String localNodeId, String modelId) throws MLException {
+        List<String> allMLNodeIdList = Arrays
+            .stream(nodeHelper.getAllNodes())
+            .filter(MLNodeUtils::isMLNode)
+            .map(DiscoveryNode::getId)
+            .collect(Collectors.toList());
+
+        if (!allMLNodeIdList.contains(localNodeId)) {
+            allMLNodeIdList.add(localNodeId);
+        }
+        MLLoadModelRequest mlLoadModelRequest = new MLLoadModelRequest(modelId, allMLNodeIdList.toArray(new String[] {}), false, false);
+
+        client
+            .execute(
+                MLLoadModelAction.INSTANCE,
+                mlLoadModelRequest,
+                ActionListener
+                    .wrap(response -> log.info("the model {} is auto reloading under the node {} ", modelId, localNodeId), exception -> {
+                        log.error("fail to reload model " + modelId + " under the node " + localNodeId + "\nthe reason is: " + exception);
+                        throw new MLException(
+                            "fail to reload model " + modelId + " under the node " + localNodeId + "\nthe reason is: " + exception
+                        );
+                    })
+            );
+    }
+
+    /**
+     * query task index, and get the result of "task_type"="LOAD_MODEL" and "state"="COMPLETED" and
+     * "worker_node" match nodeId
+     *
+     * @param localNodeId one of query condition
+     */
+    @VisibleForTesting
+    void queryTask(String localNodeId, ActionListener<SearchResponse> searchResponseActionListener) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().from(0).size(1);
```

Comment:
This query only return the latest load model task. If user have 3 models, this query only return 1 latest load model task for 1 model, the other 2 models' tasks won't be returned. So we can't reload all 3 models, just reload 1 model. Is that correct?